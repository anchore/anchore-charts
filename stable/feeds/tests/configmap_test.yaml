suite: ConfigMap Tests
templates:
  - configmap.yaml
  - envvars_configmap.yaml
values:
  - values.yaml
release:
  name: test-release
  namespace: test-namespace
chart:
  version: 9.9.9
  appVersion: 9.9.9
tests:
  - it: should render the configmaps
    asserts:
      - matchSnapshot: {}

  - it: should render grype url correctly if .Values.url is set
    template: envvars_configmap.yaml
    set:
      url: https://something/v2/my-grype/something
    asserts:
      - matchRegex:
          path: data["ANCHORE_FEEDS_EXTERNAL_URL"]
          pattern: https://something/v2/$

  - it: should render grype url correctly if .Values.url is set to a v1 url
    template: envvars_configmap.yaml
    set:
      url: https://something/v2/my-grype/something
    asserts:
      - matchRegex:
          path: data["ANCHORE_FEEDS_EXTERNAL_URL"]
          pattern: https://something/v2/$

  - it: should render the default-unset string for the ANCHORE_GITHUB_TOKEN rather than "null"
    template: envvars_configmap.yaml
    asserts:
      - equal:
          path: data.ANCHORE_GITHUB_TOKEN
          value: "default-unset"

  - it: should render the default-unset string in envvar even if GH token is set, since that is covered by the secrets
    template: envvars_configmap.yaml
    set:
      feeds.drivers.github.enabled: true
      feeds.drivers.github.token: foobar
    asserts:
      - equal:
          path: data.ANCHORE_GITHUB_TOKEN
          value: "default-unset"