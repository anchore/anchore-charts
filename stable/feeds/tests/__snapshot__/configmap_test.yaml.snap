should render the configmaps:
  1: |
    apiVersion: v1
    data:
      config.yaml: |
        # Anchore Feeds Service Configuration File, mounted from a configmap
        #
        service_dir: ${ANCHORE_SERVICE_DIR}
        tmp_dir: ${ANCHORE_FEEDS_TMP_DIR}
        log_level: ${ANCHORE_LOG_LEVEL}

        host_id: "${ANCHORE_HOST_ID}"
        internal_ssl_verify: ${ANCHORE_INTERNAL_SSL_VERIFY}

        global_client_connect_timeout: ${ANCHORE_GLOBAL_CLIENT_CONNECT_TIMEOUT}
        global_client_read_timeout: ${ANCHORE_GLOBAL_CLIENT_READ_TIMEOUT}
        server_request_timeout_seconds: ${ANCHORE_GLOBAL_SERVER_REQUEST_TIMEOUT_SEC}

        license_file: ${ANCHORE_LICENSE_FILE}
        auto_restart_services: false

        max_source_import_size_mb: ${ANCHORE_MAX_IMPORT_SOURCE_SIZE_MB}
        max_import_content_size_mb: ${ANCHORE_MAX_IMPORT_CONTENT_SIZE_MB}
        max_compressed_image_size_mb: ${ANCHORE_MAX_COMPRESSED_IMAGE_SIZE_MB}

        metrics:
          enabled: ${ANCHORE_ENABLE_METRICS}
          auth_disabled: true

        keys:
          secret: ${ANCHORE_SAML_SECRET}
          public_key_path: ${ANCHORE_AUTH_PRIVKEY}
          private_key_path: ${ANCHORE_AUTH_PUBKEY}

        user_authentication:
          hashed_passwords: ${ANCHORE_AUTH_ENABLE_HASHED_PASSWORDS}
          sso_require_existing_users: ${ANCHORE_SSO_REQUIRES_EXISTING_USERS}
          oauth:
            enabled: ${ANCHORE_OAUTH_ENABLED}
            default_token_expiration_seconds: ${ANCHORE_OAUTH_TOKEN_EXPIRATION}
            refresh_token_expiration_seconds: ${ANCHORE_OAUTH_REFRESH_TOKEN_EXPIRATION}

        credentials:
          database:
            user: "${ANCHORE_FEEDS_DB_USER}"
            password: "${ANCHORE_FEEDS_DB_PASSWORD}"
            host: "${ANCHORE_FEEDS_DB_HOST}"
            port: "${ANCHORE_FEEDS_DB_PORT}"
            name: "${ANCHORE_FEEDS_DB_NAME}"
            db_connect_args:
              timeout: ${ANCHORE_FEEDS_DB_TIMEOUT}
              ssl: ${ANCHORE_FEEDS_DB_SSL}
            db_pool_size: ${ANCHORE_FEEDS_DB_POOL_SIZE}
            db_pool_max_overflow: ${ANCHORE_FEEDS_DB_POOL_MAX_OVERFLOW}

        services:
          feeds:
            enabled: true
            require_auth: true
            endpoint_hostname: ${ANCHORE_ENDPOINT_HOSTNAME}
            listen: '0.0.0.0'
            port: ${ANCHORE_PORT}
            max_request_threads: ${ANCHORE_MAX_REQUEST_THREADS}
            ssl_enable: ${ANCHORE_SSL_ENABLED}
            ssl_cert: ${ANCHORE_SSL_CERT}
            ssl_key: ${ANCHORE_SSL_KEY}
            cycle_timers:
              driver_sync: 7200
            local_workspace: ${ANCHORE_FEEDS_LOCAL_WORKSPACE}
            workspace_preload:
              enabled: ${ANCHORE_FEEDS_WORKSPACE_PRELOAD}
              workspace_preload_file: ${ANCHORE_FEEDS_LOCAL_WORKSPACE}/data.tar.gz
            api_only: ${ANCHORE_FEEDS_API_ONLY}
            drivers:
              amzn:
                enabled: ${ANCHORE_FEEDS_DRIVER_AMAZON_ENABLED}
              alpine:
                enabled: ${ANCHORE_FEEDS_DRIVER_ALPINE_ENABLED}
              centos:
                enabled: false
              debian:
                enabled: ${ANCHORE_FEEDS_DRIVER_DEBIAN_ENABLED}
              ol:
                enabled: ${ANCHORE_FEEDS_DRIVER_OL_ENABLED}
              ubuntu:
                enabled: ${ANCHORE_FEEDS_DRIVER_UBUNTU_ENABLED}
                git_url: ${ANCHORE_FEEDS_DRIVER_UBUNTU_URL}
                git_branch: ${ANCHORE_FEEDS_DRIVER_UBUNTU_BRANCH}
              rhel:
                enabled: ${ANCHORE_FEEDS_DRIVER_RHEL_ENABLED}
                concurrency: ${ANCHORE_FEEDS_DRIVER_RHEL_CONCURRENCY}
              npm:
                enabled: ${ANCHORE_FEEDS_DRIVER_NPM_ENABLED}
              gem:
                enabled: ${ANCHORE_FEEDS_DRIVER_GEM_ENABLED}
              nvdv2:
                enabled: ${ANCHORE_FEEDS_DRIVER_NVDV2_ENABLED}
                api_key: ${ANCHORE_NVD_API_KEY}
              mariner:
                enabled: ${ANCHORE_FEEDS_DRIVER_MARINER_ENABLED}
              msrc:
                enabled: ${ANCHORE_FEEDS_DRIVER_MSRC_ENABLED}
              github:
                enabled: ${ANCHORE_FEEDS_DRIVER_GITHUB_ENABLED}
                token: ${ANCHORE_GITHUB_TOKEN}
              grypedb:
                enabled: ${ANCHORE_FEEDS_DRIVER_GRYPEDB_ENABLED}
                external_feeds_url: ${ANCHORE_FEEDS_EXTERNAL_URL}
                preload:
                  enabled: ${ANCHORE_FEEDS_GRYPEDB_PRELOAD_ENABLED}
                  workspace_archive_path: ${ANCHORE_FEEDS_GRYPEDB_PRELOAD_PATH}
                persist_provider_workspaces: ${ANCHORE_FEEDS_GRYPEDB_PERSIST_WORKSPACE}
                restore_provider_workspaces: ${ANCHORE_FEEDS_GRYPEDB_RESTORE_WORKSPACE}
              sles:
                enabled: ${ANCHORE_FEEDS_DRIVER_SLES_ENABLED}
              anchore_match_exclusions:
                enabled: ${ANCHORE_FEEDS_DRIVER_MATCH_EXCLUSIONS}
              wolfi:
                enabled: ${ANCHORE_FEEDS_DRIVER_WOLFI_ENABLED}
              chainguard:
                enabled: ${ANCHORE_FEEDS_DRIVER_CHAINGUARD_ENABLED}
    kind: ConfigMap
    metadata:
      annotations:
        bar: baz-annotation
        foo: bar-annotation
      labels:
        app.kubernetes.io/component: feeds
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-release-feeds
        app.kubernetes.io/part-of: anchore
        app.kubernetes.io/version: 9.9.9
        bar: baz-label
        foo: bar-label
        helm.sh/chart: feeds-9.9.9
      name: test-release-feeds
      namespace: test-namespace
  2: |
    apiVersion: v1
    data:
      ANCHORE_AUTH_ENABLE_HASHED_PASSWORDS: "false"
      ANCHORE_AUTH_PRIVKEY: "null"
      ANCHORE_AUTH_PUBKEY: "null"
      ANCHORE_CLI_URL: http://localhost:8228
      ANCHORE_CLI_USER: admin
      ANCHORE_DISABLE_METRICS_AUTH: "false"
      ANCHORE_ENABLE_METRICS: "false"
      ANCHORE_FEEDS_API_ONLY: "false"
      ANCHORE_FEEDS_DB_POOL_MAX_OVERFLOW: "100"
      ANCHORE_FEEDS_DB_POOL_SIZE: "30"
      ANCHORE_FEEDS_DB_SSL: "false"
      ANCHORE_FEEDS_DB_SSL_MODE: require
      ANCHORE_FEEDS_DB_SSL_ROOT_CERT: "null"
      ANCHORE_FEEDS_DB_TIMEOUT: "120"
      ANCHORE_FEEDS_DRIVER_ALPINE_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_AMAZON_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_CHAINGUARD_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_DEBIAN_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_GEM_ENABLED: "false"
      ANCHORE_FEEDS_DRIVER_GITHUB_ENABLED: "false"
      ANCHORE_FEEDS_DRIVER_GRYPEDB_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_MARINER_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_MATCH_EXCLUSIONS: "true"
      ANCHORE_FEEDS_DRIVER_MSRC_ENABLED: "false"
      ANCHORE_FEEDS_DRIVER_NPM_ENABLED: "false"
      ANCHORE_FEEDS_DRIVER_NVDV2_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_OL_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_RHEL_CONCURRENCY: "5"
      ANCHORE_FEEDS_DRIVER_RHEL_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_SLES_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_UBUNTU_BRANCH: master
      ANCHORE_FEEDS_DRIVER_UBUNTU_ENABLED: "true"
      ANCHORE_FEEDS_DRIVER_UBUNTU_URL: https://git.launchpad.net/ubuntu-cve-tracker
      ANCHORE_FEEDS_DRIVER_WOLFI_ENABLED: "true"
      ANCHORE_FEEDS_EXTERNAL_URL: http://test-release-feeds:8448/v2/
      ANCHORE_FEEDS_GRYPEDB_PERSIST_WORKSPACE: "true"
      ANCHORE_FEEDS_GRYPEDB_PRELOAD_ENABLED: "true"
      ANCHORE_FEEDS_GRYPEDB_PRELOAD_PATH: /preload/grype-db-workspace.tar.gz
      ANCHORE_FEEDS_GRYPEDB_RESTORE_WORKSPACE: "true"
      ANCHORE_FEEDS_LOCAL_WORKSPACE: /workspace
      ANCHORE_FEEDS_PACKAGES_ENABLED: "false"
      ANCHORE_FEEDS_TMP_DIR: /anchore_scratch
      ANCHORE_FEEDS_WORKSPACE_PRELOAD: "true"
      ANCHORE_GITHUB_TOKEN: default-unset
      ANCHORE_GLOBAL_CLIENT_CONNECT_TIMEOUT: "0"
      ANCHORE_GLOBAL_CLIENT_READ_TIMEOUT: "0"
      ANCHORE_GLOBAL_SERVER_REQUEST_TIMEOUT_SEC: "180"
      ANCHORE_INTERNAL_SSL_VERIFY: "false"
      ANCHORE_LICENSE_FILE: /home/anchore/license.yaml
      ANCHORE_LOG_LEVEL: INFO
      ANCHORE_MAX_COMPRESSED_IMAGE_SIZE_MB: "-1"
      ANCHORE_MAX_IMPORT_CONTENT_SIZE_MB: "100"
      ANCHORE_MAX_IMPORT_SOURCE_SIZE_MB: "100"
      ANCHORE_MAX_REQUEST_THREADS: "50"
      ANCHORE_NVD_API_KEY: "null"
      ANCHORE_OAUTH_ENABLED: "false"
      ANCHORE_OAUTH_REFRESH_TOKEN_EXPIRATION: "86400"
      ANCHORE_OAUTH_TOKEN_EXPIRATION: "3600"
      ANCHORE_SAML_SECRET: "null"
      ANCHORE_SERVICE_DIR: /anchore_service
      ANCHORE_SSL_CERT: "null"
      ANCHORE_SSL_ENABLED: "false"
      ANCHORE_SSL_KEY: "null"
      ANCHORE_SSO_REQUIRES_EXISTING_USERS: "false"
      ANCHORECTL_URL: http://localhost:8228
      ANCHORECTL_USERNAME: admin
    kind: ConfigMap
    metadata:
      annotations:
        bar: baz-annotation
        foo: bar-annotation
      labels:
        app.kubernetes.io/component: feeds
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-release-feeds
        app.kubernetes.io/part-of: anchore
        app.kubernetes.io/version: 9.9.9
        bar: baz-label
        foo: bar-label
        helm.sh/chart: feeds-9.9.9
      name: test-release-feeds-config-env-vars
      namespace: test-namespace
